EIPS/eip-663.md

```diff
@@ -23,32 +23,39 @@ Furthermore, implementing higher level constructs, such as functions, on top of
 
 The number of these arguments (or stack items) can easily exceed 16 and thus will require extra care from a compiler to lay them out in a way that all of them are still accessible.
 
-Introducing `SWAPN` and `DUPN` will provide an option to compilers to simplify accessing deep stack items at the price of possibly increased gas costs.
+Lastly, swapping items besides the 1st and Nth items in the stack is very important for compilers implementing stack scheduling algorithms (the analog of register allocation for stack machines), which try to minimize stack traffic given a set of variables and usage analysis.
+
+Introducing `SWAPN`, `DUPN` and `EXCHANGE` will provide an option to compilers to simplify accessing deep stack items.
 
 ## Specification
 
 We introduce two new instructions:
 
  1. `DUPN` (`0xe6`)
- 2. `SWAPN` (`0xe7`)
+ 1. `SWAPN` (`0xe7`)
+ 2. `EXCHANGE` (`0xe8`)
```

Comment:
Conflicts with current propsoal for `DATALOAD`

Reply:
Have we chosen an alternative yet? Need to fix it in the megaspec (https://github.com/ipsilon/eof/issues/43)

Reply:
proposing moving a bunch of new opcodes to the D section here: https://github.com/ipsilon/eof/pull/46

Reply:
DATALOAD is being moved to 0xd0